### PR TITLE
add output file for ffdon

### PR DESCRIPTION
**Pull request by T0biii**
according to:
https://gist.github.com/herbetom/f8eae6fde7a00c59f4333b49c189c784
we can add more outputs + filter the output

config is untested

### CODE REVIEW

**Review by awlx — DISMISSED** [review]
Looks about right? :D

**Comment by herbetom** [comment]
Not quite sure if it's right. From what i could tell filtering works on a site base and what you've listed seem to me to be domains.
So you might need to move the ffdon domains in their own site and then filter for that single ffdon site.

**Comment by herbetom** [comment]
Looks better. I don't know wheter domains can appear multiple times. But it's probably worth an attempt.

File: yanic/yanic.conf.tmpl [hunk 1]
```diff
@@ -46,6 +53,14 @@ offline_after = "10m"
 enable   = true
 path = "/srv/yanic/meshviewer.json"
 
+[[nodes.output.meshviewer-ffrgb]]
```
Comment (grische) [inline yanic/yanic.conf.tmpl:56]: I think ffrgb should be replaced with ffdon here?

Comment (DasSkelett) [reply yanic/yanic.conf.tmpl:56]: This is the identifier of the output type I believe, basically "JSON for the FF RGB version of the meshviewer", so it should be correct like this. For the existing output it specifies the same name.

File: yanic/yanic.conf.tmpl [hunk 2]
```diff
@@ -12,9 +12,16 @@ synchronize      = "1m"
 # how often request per multicast
 collect_interval = "1m"
 
+# all
 [respondd.sites.ffmuc]
 domains             = [{% for site, domains in sites.items() %}{% for domain in domains %}"{{site}}_{{ domain }}", {% endfor %}{% endfor %}]
 
+# ffdon
+[respondd.sites.ffdon]
+domains             = ["ffdon_mitte", "ffdon_nordwest", "ffdon_sued"]
```
Comment (DasSkelett) [inline yanic/yanic.conf.tmpl:21]: We could try something like this to automatically populate the sites from netbox if we ever add another one to ffdon:
```suggestion
domains             = [{% for domain in sites["ffdon"] %}"ffdon_{{ domain }}", {% endfor %}]
```

Comment (T0biii) [reply yanic/yanic.conf.tmpl]: sounds good to me

**Review by DasSkelett — APPROVED** [review]
I took the liberty to push a small commit applying the changes that Black complained about in CI so it is green.
Looks good, I'll try deploying it.

**Comment by DasSkelett** [comment]
Deployed, but the JSON is empty:
https://map.ffmuc.net/data/meshviewer-ffdon.json
https://gw04.ext.ffmuc.net/meshviewer-ffdon.json
```json
{"timestamp":"2024-02-11T15:07:29+0100","nodes":[],"links":[]}
```

So maybe the same domain in multiple sites does indeed not work. I might check the yanic code later to confirm this (or see whether it could be changed)

**Comment by DasSkelett** [comment]
Turns out yanic uses the `site_code` the node itself returns in the respondd data for filtering. Which kind of makes sense thinking about it. 
https://github.com/FreifunkBremen/yanic/blob/c19c1aed1b237480895e50f0bd50e6ff3758fadd/output/filter/site/site.go#L33-L40
(`nodeinfo.System.SiteCode` is basically only written to from the respondd returned JSON)
I'm not really seeing a good way to do this in yanic right now.

Mabye we can adjust `site_code` in the gluon domain configs for the ffdon domains, but no idea what other site effects this would have (@grische ?).

**Comment by T0biii** [comment]
maybe we can set the Paramter to use the Domaincode instead?

https://github.com/FreifunkBremen/yanic/blob/c19c1aed1b237480895e50f0bd50e6ff3758fadd/output/filter/domainassite/domainassite.go#L36

**Pull request merged**